.github/workflows/slsa-builder-go.yml

```diff
@@ -82,13 +88,25 @@ jobs:
             BUILDER_DIGEST=$(sha256sum "$BUILDER_BINARY" | awk '{print $1}')
             echo "::set-output name=go-builder-sha256::$BUILDER_DIGEST"
             echo "hash of $BUILDER_BINARY is $BUILDER_DIGEST"
+            cd -
+            cd slsa-provenance
+            go build -mod=vendor -o "$PROVENANCE_BINARY"
+            PROVENANCE_DIGEST=$(sha256sum "$PROVENANCE_BINARY" | awk '{print $1}')
+            echo "::set-output name=go-prov-sha256::$PROVENANCE_DIGEST"
+            echo "hash of $PROVENANCE_BINARY is $PROVENANCE_DIGEST"
 
       - name: Upload the builder
         uses: actions/upload-artifact@82c141cc518b40d92cc801eee768e7aafc9c2fa2 # v2.3.1
         with:
           name: "${{ env.BUILDER_BINARY }}"
           path: "build-go/${{ env.BUILDER_BINARY }}"
           retention-days: 5
+      - name: Upload the provenance binary
+        uses: actions/upload-artifact@82c141cc518b40d92cc801eee768e7aafc9c2fa2 # v2.3.1
```

Comment:
would be great to use a single instance of upload and upload several files https://github.com/actions/upload-artifact#upload-using-multiple-paths-and-exclusions
Don't do that in this PR.

How about adding `if-no-files-found: error` for all upload? Just in case

Reply:
Yeah, I think we should also be able to combine the build steps and output them inside a folder